python-avd/pyavd/_eos_designs/eos_designs_facts/uplinks.py

```diff
@@ -299,7 +299,10 @@ def _get_l2_uplink(
         if uplink_native_vlan := get(self.shared_utils.switch_data_combined, "uplink_native_vlan"):
             uplink["native_vlan"] = uplink_native_vlan
 
-        if self._short_esi is not None:
+        if self.shared_utils.mlag_role == "secondary":
+            # On the MLAG Secondary use short-esi from MLAG primary
+            uplink["peer_short_esi"] = self.shared_utils.mlag_peer_facts._short_esi
```

Comment:
To keep it consistent with the regular case.
```suggestion
            if (peer_short_esi := self.shared_utils.mlag_peer_facts._short_esi) is not None:
                uplink["peer_short_esi"] = peer_short_esi
```